tools/gen_compilation_database.py

```diff
@@ -1,18 +1,47 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
```

Comment:
I'm kind of unhappy that we have to manually sync these files. Do you think we could move the common CI and dev tooling support files to an additional repository, e.g. https://github.com/envoyproxy/envoy-build-tools?

This is non-actionable for this review, but I'm hoping between you, me and @lizan we can figure out a good story here, as I would like CI and dev tools to not create undue burden on adding new envoyproxy org repos.